include/llarp/dns_rectypes.hpp

```diff
@@ -0,0 +1,114 @@
+#ifndef LIBLLARP_DNS_REC_TYPES_HPP
+#define LIBLLARP_DNS_REC_TYPES_HPP
+
+#include <vector>
+#include <llarp/buffer.h>  // for byte_t
+#include <llarp/net.hpp>   // for llarp::Addr , llarp::huint32_t
+
+namespace llarp
+{
+  namespace dns
+  {
+    struct record
+    {
+      virtual ~record(){};
+      record()
+      {
+      }
+
+      virtual bool
+      parse(std::vector< byte_t > bytes);
+
+      virtual std::vector< byte_t >
+      to_bytes();
```

Comment:
`= 0;` here too.

Reply:
what does = 0 mean?

Reply:
it means it's a pure virtual function with no implementation. meaning the subtype HAS to implement the function in order for it to be constructed.